src/models/CreateAndRunHelper.ts

```diff
@@ -0,0 +1,326 @@
+import { isNullOrUndefined } from "util";
+import { TestModel, AppComponents, ExistingParams, PassFailMetric, PassFailServerMetric, SecretMetadata, AppComponentDefinition, InputArtifacts, ServerMetricConfig, TestRunModel } from "./PayloadModels";
+import { YamlConfig } from "./TaskModels";
+import { ConditionEnumToSignMap, RunTimeParams, ValidConditionsEnumValuesList } from "./UtilModels";
+import * as Util from "./util";
+import * as InputConstants from "./InputConstants";
+import { OutputVariableName } from "./constants";
+import * as CoreUtils from './CoreUtils';
+
+export function addExistingParameters(testObj: TestModel, appcomponents: AppComponents | null): ExistingParams {
+    let existingParams: ExistingParams = { secrets: {}, env: {}, passFailCriteria: {}, passFailServerMetrics: {}, appComponents: new Map() };
+    if(!isNullOrUndefined(testObj.passFailCriteria) && !isNullOrUndefined(testObj.passFailCriteria.passFailMetrics))
+        existingParams.passFailCriteria = testObj.passFailCriteria.passFailMetrics;
+    if(!isNullOrUndefined(testObj.passFailCriteria) && !isNullOrUndefined(testObj.passFailCriteria.passFailServerMetrics))
+        existingParams.passFailServerMetrics = testObj.passFailCriteria.passFailServerMetrics;
+    if(!isNullOrUndefined(testObj.secrets)){
+        existingParams.secrets = testObj.secrets;
+    }
+    if(!isNullOrUndefined(testObj.environmentVariables)){
+        existingParams.env = testObj.environmentVariables;
+    }
```

Comment:
nit: Can we have some consistency here? At some places its:

```ts
if (condition)
    statement;
```

and at some places it is:

```ts
if (condition) { 
   statement;
}
```

Preference wise I would love to not omit braces